### DISCUSSION TITLE
Pydantic, would it be worthwhile for us to use it?

### DISCUSSION THREAD

**Discussion by JohnVillalovos** [python-gitlab/python-gitlab#1672] [Ideas]
Using [pydantic](https://pydantic-docs.helpmanual.io/) would seem to be a good fit for python-gitlab as it helps to check that the data we are receiving is what we expect to receive.

Any thoughts on start using pydantic in the code base?

https://pydantic-docs.helpmanual.io/

**Comment by nejch** [comment]
If I get it right from a quick glance, this would mostly be a drop-in and we'd just need to extend `BaseModel` most of the time to get runtime validation?

If so, would it be an option to add it to `extra_requires` as a start - and have it as a `pip install python-gitlab[pydantic]` option for people who'd like to have this? We'd just need some convenient try/except import somewhere, I assume, with a dummy/fallback `BaseModel` object or so.

**Reply by JohnVillalovos** [reply]
I don't think it would be an "optional" thing. It would be a library we start using to help with data validation.

It would be overly complicated if we had multiple classes doing the same thing. One with pydantic support and one without. IMHO.
